library/std/src/f128/tests.rs

```diff
@@ -2,7 +2,9 @@
 #![cfg(reliable_f128)]
```

Comment:
This doesn't seem to work. I disregarded that as I wanted to have a minimal PR. Should I remove this, move the conditional compilation to the `mod tests;` statement (but that's even easier for developers to miss), or use a `mod` statement within this `tests` module with a proper single conditional compilation statement?

Reply:
That config is set here https://github.com/rust-lang/rust/blob/4d215e2426d52ca8d1af166d5f6b5e172afbff67/library/std/build.rs#L120-L143 and at this point should really only cover platforms where LLVM crashes or ABI problems lead to buggy results. `reliable_f128_math` is the gate that should cover anything that relies on `libm` symbols. I think we just didn't catch this because `fmodl` is available on many platforms even when the rest of the math symbols might not be.

So moving only calls to functions that rely on `f128` `libm` to `reliable_f128_math` should be the needed correction (suggested by @beetrees below), everything else should continue to work on SGX and other platforms.